### ISSUE TITLE
[Bug]:milvus集群数据存储占用大量磁盘空间

### ISSUE THREAD

**Issue by hukang6677** [milvus-io/milvus#19049]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.1.0
- Deployment mode(standalone or cluster):cluster k8s
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): Ubuntu 
- CPU/Memory: 128G
- GPU: 
- Others:
```

### Current Behavior

milvus集群数据存储占用大量磁盘空间，120万256维数据，索引是IVF_SQ8，milvus使用默认配置。主要是四个minio占用了
![lQLPJxapCgcOzIXM7M0EZbBZU0LST5pg0QMWBSRtQNIA_1125_236](https://user-images.githubusercontent.com/102129512/188568253-59eafa7c-af8d-42c7-8c66-498044cf71f3.png)

### Steps To Reproduce

```markdown
1 在k8s中使用helm搭建milvus集群，一个master,两个node（node1内存110G，node2内存240G）
2 初始node1上的pod较多，我手动驱逐了node1上的pod,转移到node2上去
3 创建collection,loadcollection,creatindex,256维，索引是IVF_SQ8。然后往collection中插入数据，每5秒 1000条数据
4 数据量达到120万左右
```

### Milvus Log

[milvus-log.tar.gz](https://github.com/milvus-io/milvus/files/9494235/milvus-log.tar.gz)

**Comment by LoveEachDay** [comment]
@hukang6677  From the attached log:
```
[2022/09/05 13:54:28.052 +00:00] [WARN] [storage/minio_chunk_manager.go:326] ["failed to list with prefix"] [prefix=file/insert_log] [requestID=] [error="Get \"http://my-release-minio:9000/milvus-bucket/?continuation-token=ZmlsZS9pbnNlcnRfbG9nLzQzNTc3OTYzNDg1MzgzODg0OS80MzU3Nzk2MzQ4NTM4Mzg4NTAvNDM1NzgyMjMzMzQzMTMxNjQ5LzEwOC80MzU3ODIyNDM1NTM5MDI2MDFbbWluaW9fY2FjaGU6djIsaWQ6ZWYzOGJmNjctODI5My00MmI4LWEwMWMtNjFlZGVmMGRjYzY2LHA6MCxzOjBd&delimiter=&encoding-type=url&fetch-owner=true&list-type=2&prefix=file%2Finsert_log\": net/http: timeout awaiting response headers"]
[2022/09/05 13:54:28.053 +00:00] [ERROR] [datacoord/garbage_collector.go:131] ["gc listWithPrefix error"] [error="Get \"http://my-release-minio:9000/milvus-bucket/?continuation-token=ZmlsZS9pbnNlcnRfbG9nLzQzNTc3OTYzNDg1MzgzODg0OS80MzU3Nzk2MzQ4NTM4Mzg4NTAvNDM1NzgyMjMzMzQzMTMxNjQ5LzEwOC80MzU3ODIyNDM1NTM5MDI2MDFbbWluaW9fY2FjaGU6djIsaWQ6ZWYzOGJmNjctODI5My00MmI4LWEwMWMtNjFlZGVmMGRjYzY2LHA6MCxzOjBd&delimiter=&encoding-type=url&fetch-owner=true&list-type=2&prefix=file%2Finsert_log\": net/http: timeout awaiting response headers"] [stack="github.com/milvus-io/milvus/internal/datacoord.(*garbageCollector).scan\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/garbage_collector.go:131\ngithub.com/milvus-io/milvus/internal/datacoord.(*garbageCollector).work\n\t/go/src/github.com/milvus-io/milvus/internal/datacoord/garbage_collector.go:96"]
```

The reason for storage increasing is that compaction will compact small segments into bigger segments, and if gc is failed, the old segments cannot be removed.  And we can see gc is failed because minio `listWithPrefix` timeout. 

Could you provide some information about your insertion?  Or may be you can provide your insert script ? From your description above, you insert 1000 vector into collection every 5 seconds, did you called the `num_entities` on the collection?

**Comment by hukang6677** [comment]
1. minio的 listWithPrefix timeout的原因要如何设置合理值。
2. Milvus关于num_entities 参数的资料在哪可以看到，该参数是否也是针对数据插入场景做调整。

**Comment by hukang6677** [comment]
插入脚本如下：只是简单的做了个定时任务执行insert操作，应该没有使用num_entities

 @Scheduled(cron = "0/5 * * * * ?")
    public void insertMilvusTask() {
        System.out.println("定时任务开始-----------");
        insertMilvus();

    }

    public void insertMilvus() {
        Random ran = new Random();
        List<String> vehicle_id_array = new ArrayList<>();
        List<List<Float>> vehicle_feature_array = new ArrayList<>();
        List<Long> pass_time_array = new ArrayList<>();
        List<Integer> road_id_array = new ArrayList<>();
        List<Long> video_uuid_array = new ArrayList<>();
        List<Integer> vehicle_type_array = new ArrayList<>();
        List<Integer> vehicle_color_array = new ArrayList<>();
        List<String> plate_num_array = new ArrayList<>();
        List<Integer> plate_color_array = new ArrayList<>();
        List<Integer> speed_array = new ArrayList<>();

        for (long i = 1; i <= 1000; ++i) {
            vehicle_id_array.add(UUID.randomUUID().toString().replace("-", ""));
            List<Float> vector = new ArrayList<>();
            for (int k = 0; k < 256; ++k) {
                vector.add(ran.nextFloat());
            }
            vehicle_feature_array.add(vector);
            pass_time_array.add(1659075494123L);
            road_id_array.add(123);
            video_uuid_array.add(456L);
            vehicle_type_array.add(1);
            vehicle_color_array.add(2);
            plate_num_array.add("湘A.123AB");
            plate_color_array.add(3);
            speed_array.add(90);
        }
        List<InsertParam.Field> fields = new ArrayList<>();
        fields.add(new InsertParam.Field("vehicle_id", vehicle_id_array));
        fields.add(new InsertParam.Field("vehicle_feature", vehicle_feature_array));
        fields.add(new InsertParam.Field("pass_time", pass_time_array));
        fields.add(new InsertParam.Field("road_id", road_id_array));
        fields.add(new InsertParam.Field("video_uuid", video_uuid_array));
        fields.add(new InsertParam.Field("vehicle_type", vehicle_type_array));
        fields.add(new InsertParam.Field("vehicle_color", vehicle_color_array));
        fields.add(new InsertParam.Field("plate_num", plate_num_array));
        fields.add(new InsertParam.Field("plate_color", plate_color_array));
        fields.add(new InsertParam.Field("speed", speed_array));
        InsertParam insertParam = InsertParam.newBuilder()
                .withCollectionName("vehicle_search20220905")
//                .withPartitionName("test")
                .withFields(fields)
                .build();
        long run = System.currentTimeMillis();
        milvusClient.insert(insertParam);
        long end = System.currentTimeMillis();
        System.out.println("数据插入耗时：" + (end - run));
    }

**Comment by hukang6677** [comment]
官方文档上helm安装milvus集群上StorageClass 的回收策略是 Delete。我的StorageClass回收策略是 "回收后保留"，请问下跟这个有关系吗？

![597588BD-D90D-4958-BD29-4771C63AE610](https://user-images.githubusercontent.com/102129512/188790409-b07c9e82-0a16-41a9-9f84-35ffd396b030.png)

**Comment by LoveEachDay** [comment]
@hukang6677 We're trying to fix this timeout in PR #19096. You can try to upgrade your cluster with new image `milvusdb/milvus:2.1.0-20220908-a99a966`.

**Comment by xiaofan-luan** [comment]
upgrade to 2.1.4 would solve the issue
